cocos/2d/assembler/label/text-processing.ts

```diff
@@ -560,7 +563,12 @@ export class TextProcessing {
                 }
                 if (cclegacy.director.root && cclegacy.director.root.batcher2D) {
                     if (JSB) {
-                        cclegacy.director.root.batcher2D._releaseDescriptorSetCache(tex.getGFXTexture(), tex.getGFXSampler());
+                        // NOTE: Release the old descriptor set cache referenced by old gfx texture and sampler.
+                        // We should not release the new generated `tex.getGFXTexture()` and `tex.getGFXSampler()`
+                        // since `tex.reset(...)` will reset `gfxTexture` and `gfxSampler`.
+                        // The other non-JSB branch uses `tex.getHash()` which returns the hash value that will not be
+                        // changed after `tex.reset(...)`, so there will be no problems for non-JSB branch.
+                        cclegacy.director.root.batcher2D._releaseDescriptorSetCache(oldGfxTexture, oldGfxSampler);
```

Comment:
If so, can use the same logic in for all platforms?

Reply:
It probably could, but I don't know the reason why it was designed like this.
This code was written in v3.6.0: https://github.com/cocos/cocos-engine/pull/11592
So currently, we just fix the bug and don't refactor it.